bbq/vm/vm.go

```diff
@@ -465,9 +471,11 @@ func opInvokeDynamic(vm *VM, ins opcode.InstructionInvokeDynamic) {
 	var functionValue = vm.lookupFunction(compositeType.Location, qualifiedFuncName)
 
 	parameterCount := int(functionValue.Function.ParameterCount)
-	arguments := vm.stack[stackHeight-parameterCount:]
+	arguments := vm.peekN(parameterCount)
 	vm.pushCallFrame(functionValue, arguments)
-	vm.dropN(parameterCount)
+	vm.dropN(len(arguments))
+
+	// TODO: drop the receiver?
```

Comment:
`parameterCount` already includes the receiver. see: https://github.com/onflow/cadence/blob/6ce5b49adbfc6c730589286b8f512e68d8642f39/bbq/compiler/compiler.go#L1208-L1210

So we don't have to explicitly drop it.

On a related note: All tests do this sanity check on the stack size, at the end of the execution, just to make sure no garbage values are left on stack: https://github.com/onflow/cadence/blob/6ce5b49adbfc6c730589286b8f512e68d8642f39/bbq/vm/test/utils.go#L631-L634

Reply:
Ah, nice! 👌  Thanks for clarifying!

```suggestion
	// We do not need to drop the receiver, as the parameter count given in the instruction already includes it
```